fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PartitionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,159 @@
+        'col2'      | ['2018-01-01/2018-02-01']  | []                                                 | [col1: ['2018-01-01/2018-02-01']]                                  | "different columns with one column being empty"
+        'col1'      | ['2018-01-01/2018-02-01']  | []                                                 | [col1: ['2018-01-01/2018-02-01']]                                  | "the same columns with one column being empty"
+        'col2'      | ['2018-01-01/2018-02-01']  | ['2018-01-01/2018-02-01']                          | [col1: ['2018-01-01/2018-02-01'], col2: ['2018-01-01/2018-02-01']] | "different columns with same list of intervals"
+        'col1'      | ['2018-01-01/2018-02-01']  | ['2018-01-01/2018-02-01']                          | [col1: ['2018-01-01/2018-02-01']]                                  | "the same columns with same list of intervals"
+        'col1'      | ['2018-01-01/2018-02-01']  | ['2018-02-01/2018-03-01', '2019-01-01/2019-02-01'] | [col1: ['2018-01-01/2018-03-01', '2019-01-01/2019-02-01']]         | "the same columns with intervals that need to be merged"
+    }
+
+    def "getAllAvailableIntervals returns intervals of only configured columns"() {
+        given:
+        column1 = new Column('col1')
+        column2 = new Column('col2')
+
+        availability1.getAllAvailableIntervals() >> [
+                (column1): ['2018-01-01/2018-02-01']
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (column2): ['2018-01-01/2018-02-01']
+        ]
+
+        partitionAvailability = new PartitionAvailability(
+                [availability1, availability2] as Set,
+                [column1] as Set,
+                {}
+        )
+
+        when:
+        Map<Column, List<Interval>> actual = partitionAvailability.getAllAvailableIntervals()
+
+        then:
+        actual.containsKey(column1)
+        !actual.containsKey(column2)
+    }
+
+    def "getAvailableIntervals returns intervals in intersection when intervals have #reason"() {
+        given:
+        availability1.getAvailableIntervals(_ as DataSourceConstraint) >> new SimplifiedIntervalList(
+                availableIntervals1.collect{it -> new Interval(it)} as Set
+        )
+        availability2.getAvailableIntervals(_ as DataSourceConstraint) >> new SimplifiedIntervalList(
+                availableIntervals2.collect{it -> new Interval(it)} as Set
+        )
+
+        Function<DataSourceConstraint, Set<Availability>> partitionFunction = Mock(Function.class)
+        partitionFunction.apply(_ as DataSourceConstraint) >> Sets.newHashSet(availability1, availability2)
+
+        partitionAvailability = new PartitionAvailability(
+                [availability1, availability2] as Set,
+                Collections.emptySet(),
+                partitionFunction
+        )
+
+        expect:
+        partitionAvailability.getAvailableIntervals(Mock(DataSourceConstraint)) == new SimplifiedIntervalList(
```

Comment:
Do we have a test that verifies the behavior of the partition stuff? In particular, I would expect a test that has many availabilities, and a partition function that maps to a subset, and the resulting available intervals includes only the subset of available intervals, not the full set of available intervals.

This test _looks_ like it might be doing it, but I don't think it is...